demos/es7/math-bot-es7.js

```diff
@@ -35,24 +35,20 @@ async function main() {
     const paperkey = process.env.KB_PAPERKEY
     await bot.init(username, paperkey)
     console.log('I am me!', bot.myInfo().username, bot.myInfo().devicename)
+    console.log('Beginning watch for new messages.')
+    console.log(`Tell anyone to send a message to ${bot.myInfo().username} starting with '/math '`)
 
     const onMessage = async message => {
-      try {
-        if (message.content.type === 'text') {
-          const prefix = message.content.text.body.slice(0, 6)
-          if (prefix === '/math ') {
-            const reply = {body: msgReply(message.content.text.body.slice(6))}
-            await bot.chat.send(message.channel, reply)
-          }
+      if (message.content.type === 'text') {
+        const prefix = message.content.text.body.slice(0, 6)
+        if (prefix === '/math ') {
+          const reply = {body: msgReply(message.content.text.body.slice(6))}
+          await bot.chat.send(message.channel, reply)
```

Comment:
yup! previously we had two handlers for each of those cases. There's also the possibility that we correctly parse the json, but the json contains an error message from the service. These errors are also now passed along. 

I actually don't handle the case of the spawned process exiting since I thought that we specifically wanted to drop those errors – I'll play around to see if we can conditionally throw an error only if the process isn't passed a SIGTERM.